src/java/voldemort/server/VoldemortServer.java

```diff
@@ -382,7 +382,15 @@ protected void startInner() throws VoldemortException {
         JNAUtils.tryMlockall();
         logger.info("Starting " + basicServices.size() + " services.");
         long start = System.currentTimeMillis();
-        boolean goOnline = true;
+
+        boolean goOnline;
+        if (getMetadataStore().getServerStateUnlocked() == MetadataStore.VoldemortState.NORMAL_SERVER)
+            goOnline = true;
+        else if (getMetadataStore().getServerStateUnlocked() == MetadataStore.VoldemortState.OFFLINE_SERVER)
+            goOnline = false;
+        else
+            throw new VoldemortException("Invalid Server state. Neither online nor offline found");
```

Comment:
The server has 3 states, online, offline and rebalancing. This makes the server not to start when that is in the Rebalancing mode. I would like to have only 2 cases here, offline and everything else.